dash/backend/src/modules/falco/dao/falco.dao.ts

```diff
@@ -102,26 +104,113 @@ export class FalcoDao {
+        }
+    }
+
+    async getFalcoCsvLogs(
+        clusterId: number,
+        priorities?: string [],
+        orderBy?: string,
+        startDate?: string,
+        endDate?: string,
+        namespace?: string,
+        pod?: string,
+        image?: string,
+    ): Promise<{ csvLogList: FalcoDto[] }> {
+        const knex = await this.databaseService.getConnection();
+
+        let query = knex.select()
+            .from('project_falco_logs')
+            .where('cluster_id', clusterId);
+
+        if (priorities) {
+            query = query.whereIn('level', priorities);
+        }
+        if (orderBy == 'Priority Desc' || orderBy =='Priority Asc' ||  orderBy =='Date Desc'||  orderBy =='Date Asc' ||  orderBy == null ||  orderBy == undefined) {
+            switch (orderBy) {
+                case 'Priority Desc':
+                    query = query.orderByRaw(
+                        'CASE ' +
+                        ' WHEN level = \'Emergency\' then 1' +
+                        ' WHEN level = \'Alert\' then 2' +
+                        ' WHEN level = \'Critical\' then 3' +
+                        ' WHEN level = \'Error\' then 4' +
+                        ' WHEN level = \'Warning\' then 5' +
+                        ' WHEN level = \'Notice\' then 6' +
+                        ' WHEN level = \'Informational\' then 7' +
+                        ' WHEN level = \'Debug\' then 8' +
+                        'END'
+                    );
+                    break;
+                case 'Priority Asc':
+                    query = query.orderByRaw(
+                        'CASE ' +
+                        ' WHEN level = \'Debug\' then 1' +
+                        ' WHEN level = \'Informational\' then 2' +
+                        ' WHEN level = \'Notice\' then 3' +
+                        ' WHEN level = \'Warning\' then 4' +
+                        ' WHEN level = \'Error\' then 5' +
+                        ' WHEN level = \'Critical\' then 6' +
+                        ' WHEN level = \'Alert\' then 7' +
+                        ' WHEN level = \'Emergency\' then 8' +
+                        'END'
+                    );
+                    break;
+                case 'Date Desc':
+                    query = query.orderBy([{column: 'creation_timestamp', order: 'desc'}]);
+                    break;
+                case 'Date Asc':
+                    query = query.orderBy([{column: 'creation_timestamp', order: 'asc'}]);
+                    break;
+                default:
+                    query = query.orderBy([{column: 'id', order: 'desc'}]);
+            }
+        }
+
+        if (startDate) {
+            query = query.andWhere('calendar_date', '>=', startDate);
+        }
+        if (endDate) {
+            query = query.andWhere('calendar_date', '<=', endDate);
+        }
+
+        if (namespace) {
+            query = query.whereRaw(`namespace LIKE ?`, [`%${namespace.trim()}%`]);
```

Comment:
whenever using raw queries, we need to be careful to prevent sql injection. I think that in this case, sql injection would be possible. We should use regex to ensure that namespace is alphanumeric OR dashes (capital or lowercase letters, numbers, or dashes)

Reply:
For example, if someone passed in a namespace of the following, it would be bad: 

'; DELETE FROM user; SELECT * from project_falco_logs where ''='

Reply:
Did some googling and the way you are using string interpolation is actually correct. 

https://stackoverflow.com/questions/50327070/how-do-i-escape-in-knex-where-like-query